src/keyfile.c

```diff
@@ -1268,6 +1287,16 @@ static gboolean open_session_file(gchar **tmp, guint len)
 	unescaped_filename = g_uri_unescape_string(tmp[7], NULL);
 	locale_filename = utils_get_locale_from_utf8(unescaped_filename);
 
+	if (!g_path_is_absolute(locale_filename) && app->project)
```

Comment:
Could you add a comment to note that this block unpacks relative paths for project session docs?  (So people looking at it later will know what it's doing without having to stare at the code.)

Reply:
What's supposed to happen when locale_filename is relative, but a project is not open?  Does the document just get dropped?

Reply:
> What's supposed to happen when locale_filename is relative, but a project is not open? Does the document just get dropped?

That should simply never happen, this patch writes relative paths only to project files, the global session file is unaffected.

Reply:
> That should simply never happen...

Even though this PR doesn't write relative paths in certain ways, session and project files can be edited externally or another PR may change it.  Right now, it looks like such files would be dropped.  It would be nice to know whether that behavior is intentional in case someone wants to change it someday (eg, as part of the config/session split).

Reply:
There's nothing else you could do in this case than drop it. This is no change from the current state - if someone changes a path in the file to something invalid, there's no way to open it.

Reply:
> There's nothing else you could do in this case than drop it.

Relative path using the pwd, startup path (set in preferences), config path (`-c` option), session path, etc.

Reply:
In my opinion the code itself says what it does clearly enough - it doesn't drop any files by itself. The whole logic is preserved as before - if a file from the project isn't found, it just isn't opened. The code here just converts relative paths to absolute paths using project base directory.

And yeah, I guess you know my opinion about adding a config option for every single nonsense by now ;-).